src/libs/ReportActionsUtils.ts

```diff
@@ -314,7 +318,16 @@ function shouldReportActionBeVisible(reportAction: OnyxEntry<ReportAction>, key:
 
     // Ignore closed action here since we're already displaying a footer that explains why the report was closed
     if (reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.CLOSED) {
-        return false;
+        if (!allReports) {
+            return false;
+        }
+        const expenseReport = allReports[reportAction?.originalMessage?.IOUReportID];
+        return (
+            expenseReport !== null &&
```

Comment:
Kinda looks like maybe the reportAction is not picking up the types from here properly...

https://github.com/Expensify/App/blob/dee9b3110a7ebe0aa250c3f966d6a69925d916f4/src/types/onyx/ReportAction.ts#L109-L109

That's an "intersection type" so it should merge those two together...

And if you look at `OriginalMessageIOU` it should include `IOUReportID`

https://github.com/Expensify/App/blob/dee9b3110a7ebe0aa250c3f966d6a69925d916f4/src/types/onyx/OriginalMessage.ts#L18-L36

So, it appears that the union types from here: 

https://github.com/Expensify/App/blob/dee9b3110a7ebe0aa250c3f966d6a69925d916f4/src/types/onyx/OriginalMessage.ts#L181-L195

Are overwriting each other when the intersection type gets "merged". The one it thinks you have is this:

https://github.com/Expensify/App/blob/dee9b3110a7ebe0aa250c3f966d6a69925d916f4/src/types/onyx/OriginalMessage.ts#L90-L97

Reply:
Huh. I tried creating a `OriginalMessageDone` with my structure and didn't help, probably because the actionName is the same. I'll try editing that one and seeing if it likes that better

Reply:
Yep, that did it